plugin/src/main/java/com/google/errorprone/xplat/checker/JodaTimeClassBan.java

```diff
@@ -0,0 +1,269 @@
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.ImportTreeMatcher;
+import com.google.errorprone.bugpatterns.BugChecker.MethodInvocationTreeMatcher;
+import com.google.errorprone.bugpatterns.BugChecker.MethodTreeMatcher;
+import com.google.errorprone.bugpatterns.BugChecker.NewClassTreeMatcher;
+import com.google.errorprone.bugpatterns.BugChecker.VariableTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.ImportTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.NewClassTree;
+import com.sun.source.tree.VariableTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import com.sun.tools.javac.code.Symbol.VarSymbol;
+import com.sun.tools.javac.code.Type;
+
+/**
+ * Check for usage of Joda-Time's {@code org.joda.time.base.* org.joda.time.chrono.*
+ * org.joda.time.convert.*, org.joda.time.field.*, org.joda.time.format.*, org.joda.time.tz.*,
+ * org.joda.time.Chronology, org.joda.time.DateMidnight, org.joda.time.DateTimeComparator,
+ * org.joda.time.DateTimeField, org.joda.time.DateTimeFieldType, org.joda.time.DateTimeUtils,
+ * org.joda.time.Days, org.joda.time.DurationField, org.joda.time.DurationFieldType,
+ * org.joda.time.Hours, org.joda.time.IllegalFieldValueException, org.joda.time.IllegalInstantException,
+ * org.joda.time.JodaTimePermission, org.joda.time.Minutes, org.joda.time.MonthDay,
+ * org.joda.time.Months, org.joda.time.MutableDateTime, org.joda.time.MutableInterval,
+ * org.joda.time.MutablePeriod, org.joda.time.Partial, org.joda.time.PeriodType,
+ * org.joda.time.ReadWritableDateTime, org.joda.time.ReadWritableInstant,
+ * org.joda.time.ReadWritableInterval, org.joda.time.ReadWritablePeriod org.joda.time.Seconds,
+ * org.joda.time.TimeOfDay org.joda.time.Weeks org.joda.time.YearMonthDay org.joda.time.Years}.
+ * These calls are banned due to their incompatibility with cross platform development.
+ */
+@BugPattern(
+    name = "JodaTimeClassBan",
+    summary = "Joda Time cross platform class ban",
+    explanation =
+        "The usage of several Joda Time classes are banned from cross platform development due"
+            + " to incompatibilities. They are unsupported on the web and should also not be"
+            + " used on supported platforms.",
+    severity = ERROR)
+public class JodaTimeClassBan extends BugChecker
+    implements MethodInvocationTreeMatcher, NewClassTreeMatcher, ImportTreeMatcher,
+    VariableTreeMatcher, MethodTreeMatcher {
+
+  private static final ImmutableSet<String> PACKAGE_NAMES =
+      ImmutableSet.of(
+          "org.joda.time.base.", "org.joda.time.chrono", "org.joda.time.convert",
```

Comment:
Why this one ends with a period while others don't?

Reply:
I have added a period to the summary and left the period for the explanation, to follow the formatting of other error prone checkers.